### PR TITLE
Don't look for release notes in pre-releases

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
This makes release notes optional for pre-releases (tagged with `rc`).  Pre-release versions will simply have "pre-release $TAG" for their notes in the github release.  Builds for versions without rc in the tag will still crash if release notes are missing

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
LGTM! Maybe I'll wait for @jni to comment on before merge. I do see some advantage of being able to look at the rendered release notes before releasing, but one can't have everything. I'm curious what @jni recommends.

**Comment by tlambert03** [comment]
you can see rendering of any md file in a PR by clicking "view file" in the three dots icon in the "files changed" tab.  for example:
https://github.com/napari/napari/blob/d371b9eb43335f3e6c05de10e1b3438130380a03/docs/release/release_0_3_6.md

**Comment by sofroniewn** [comment]
> you can see rendering of any md file in a PR by clicking "view file" in the three dots icon in the "files changed" tab. for example:

yeah, ok that's fair enough. I will merge this now and press on - it is very much a two way door!!

**Comment by tlambert03** [comment]
(you can also [preview markdown in vscode](https://code.visualstudio.com/docs/languages/markdown#_markdown-preview) 😉  )

**Pull request merged**
